### PR TITLE
Fix generics on LeadDocLookup

### PR DESCRIPTION
**Pull request by nik9000**
All the warnings were upsetting me. This doesn't change behavior.

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by nik9000** [comment]
Thanks Jason!

File: core/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java [hunk 1]
```diff
@@ -67,20 +67,20 @@ public void setDocument(int docId) {
     }
 
     @Override
-    public Object get(Object key) {
+    public ScriptDocValues<?> get(Object key) {
         // assume its a string...
         String fieldName = key.toString();
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java]: This means you could get a non-null value, even for a non-null key, which I find a bit confusing given the generics say that the keys are strings? Maybe we should change it to a `Map<Object, ScriptDocValues<?>>` or stop calling `toString()` and just return `null` if the key is not a `String`?

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java]: Hmm - I think, yeah, I'll stop calling toString.

I didn't notice this comment and have already merged. I'll finish the merge and open a followup and ping you on it.

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java]: Good call @jpountz.

**Comment by nik9000** [comment]
master: f7071325c4f21e7fc7071db28afa57f6a8bcb691
5.x: ab9d98e5ed6662199dfa4d5a32a18a5aec60dbb8

**Pull request merged**